### PR TITLE
updated the user grades method to allow for additional query strings …

### PR DESCRIPTION
**Pull request by plungerman**
…to filter results by timestamp, assignment ID, or enrollment ID; the _get() method had to be updated to allow you send the query string to it; that might not be nessary if the 'limit' GET query is removed, which it should be and does not belong in the HTTP methods

### CODE REVIEW

File: schoolopy/main.py [hunk 1]
```diff
@@ -1891,15 +1893,44 @@ def get_assignment_comment(self, section_id, assignment_id, comment_id):
...
     def get_friend_requests(self, user_id):
         return [FriendRequest(raw) for raw in self._get('users/%s/requests/friends' % user_id)['request']]
 
     def get_friend_request(self, user_id, request_id):
         return FriendRequest(self._get('users/%s/requests/friends/%s' % (user_id, request_id)))
 
-
     def get_user_section_invites(self, user_id):
-        return [Invite(raw) for raw in self._get('users/%s/invites/sections' % user_id)['invite']]
+        return [Invite(raw) for raw in self._get('users/{}/invites/sections'.format(user_id))['invite']]
```
Comment (ErikBoesen) [inline schoolopy/main.py]: Seems kind of weird to only do this for one function.

File: schoolopy/main.py [hunk 2]
```diff
@@ -1891,15 +1893,44 @@ def get_assignment_comment(self, section_id, assignment_id, comment_id):
...
+                        e.g. "timestamp": 1517551200
+        assignment_id:  filter grades for a given assignment
+        enrollment_id:  filter grades for a given enrollment
+        """
+
+        query = ''
+        if query_name and query_value:
+            query = '&{}={}'.format(query_name, query_value)
+
+        return [
```
Comment (ErikBoesen) [inline schoolopy/main.py]: I wouldn't expand all this here unless we were doing it somewhere else.

File: schoolopy/main.py [hunk 3]
```diff
@@ -1891,15 +1893,44 @@ def get_assignment_comment(self, section_id, assignment_id, comment_id):
...
+        """
+        Returns a list of grades (paged).
+
+        :param section_id: ID of course's section.
+        :return: a list of grades for that section.
+
+        The following optional query strings can be appended to the path
+        to filter results:
+
+        :param query_name (timestamp, assignment_id, or enrollment_id)
```
Comment (ErikBoesen) [inline schoolopy/main.py]: This is a pretty non-standard docstring style. Make sure that your notation matches the others.

File: schoolopy/main.py [hunk 4]
```diff
@@ -1891,15 +1893,44 @@ def get_assignment_comment(self, section_id, assignment_id, comment_id):
...
+        :param query_value
+
+        timestamp:      return only grades that have been changed since
```
Comment (ErikBoesen) [inline schoolopy/main.py]: I'm not sure "query" is the best name or the best way to go about this. Why not specifically ask for a timestamp?

**Comment by ErikBoesen** [comment]
@AndrewLester Take a look

File: schoolopy/main.py [hunk 5]
```diff
@@ -1891,15 +1893,44 @@ def get_assignment_comment(self, section_id, assignment_id, comment_id):
...
+                        the given timestamp, according to the server time.
+                        e.g. "timestamp": 1517551200
+        assignment_id:  filter grades for a given assignment
+        enrollment_id:  filter grades for a given enrollment
+        """
+
+        query = ''
+        if query_name and query_value:
+            query = '&{}={}'.format(query_name, query_value)
```
Comment (AndrewLester) [inline schoolopy/main.py]: Wouldn't this only allow for one of the possible queries to be used per request?

Comment (ErikBoesen) [reply schoolopy/main.py]: Oh yeah, this too.

**Comment by ErikBoesen** [comment]
Stale, closing.

**Pull request closed without merging**